### PR TITLE
Import tests from Google V8 (Rest Parameters)

### PR DESCRIPTION
**Pull request by rwaldron**
These tests are derived from the following files within the Google V8 project:

    test/mjsunit/harmony/rest-params.js

### CODE REVIEW

File: test/language/rest-parameters/no-alias-arguments-strict.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.1
+description: >
+    no alias arguments
+flags: [onlyStrict]
+includes: [compareArray.js]
+---*/
+function strict(a, ...rest) {
```
Comment (arv) [inline test/language/rest-parameters/no-alias-arguments-strict.js]: This could be folded into the previous test if you remove the `flags:`

**Comment by arv** [comment]
LGTM

File: test/language/rest-parameters/no-alias-arguments-strict.js (hunk 1 above)
Comment (rwaldron) [inline test/language/rest-parameters/no-alias-arguments-strict.js]: Yes.

File: test/language/rest-parameters/arrow-function.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.1
+description: >
+    arrow functions
+---*/
+var fn = (a, b, ...c) => c;
+
+assert.sameValue(fn(), [], "`fn()` returns `[]`");
```
Comment (anba) [inline test/language/rest-parameters/arrow-function.js]: `assert.sameValue(...)` -> `assert(compareArray(...))`.

Comment (rwaldron) [inline test/language/rest-parameters/arrow-function.js]: Thanks! I missed this because V8 currently fails to parse line 8.

Comment (rwaldron) [inline test/language/rest-parameters/arrow-function.js]: All of these need to be fixed.

File: test/language/rest-parameters/arrow-function.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.1
+description: >arrow functions
+includes: [compareArray.js]
+---*/
+var fn = (a, b, ...c) => c;
+
+assert(compareArray(fn(), []), "`compareArray(fn(), [])` returns `true`");
```
Comment (bterlson) [inline test/language/rest-parameters/arrow-function.js]: Ugh, we probably need an assert.arrayElementsSameValue or somesuch. This assert(compareArray( boilerplate really grates on me. (Ok for now though)

**Comment by rwaldron** [comment]
> really grates on me

Samesies.

**Pull request merged**